bal-code-camp-rest-oidc/oidc-application-client
# Provide delegated jwt downstream across all microservices

This PR provides the first running scenario, deployed in local docker-compose runtime.

### Tested features:

- Login as library-user (wayne)
- List all availlable books
- Render the "borrow"-button, if the book is not borrowed
- Borrow the book and see the "return"-button
- Login as other library-user (banner)
- List all availlable books, not borrowed by wayne
- Borrow a book and see the "return"-button
- Login as wayne again
- Return all borrowed books
- Test all SwaggerUI's added by the previous PR from @ArthurNeudeck 
- Login via GitHub-Account
- See your Name in the Library-Client site
- See the Borrow-Buttons as you should be in the role "library_user".

### Known Issues

- Logout-Feature doesn't work. Logout by delete all sessions in Keycloak and remove cookies.
- Deploying in local docker-compose environment requiers a bunch of memory. I had to set the memory-setting up to 4G.
- Do not forget to export the local ipv4 address to KEYCLOAK_HOST env-var to start the scenario via docker-compose.
- Login via Github is working, if and only if you run keycloak only in docker-compose, and start the library-services via IDE/CLI outside Docker. Configure the env.-Variable by export KEYCLOAK_HOST=localhost before start the docker-compose.yml
